File: 2016m./uzd1./sprendimai/uzd1.py
def getMaxBackpackWeight():
    data = open("U1.txt", "r")

    maxWeight = 0

    for i in range(int(next(data))):
        weight = int(next(data))
        if(maxWeight < weight):
            maxWeight = weight

    data.close()

    return maxWeight

def countBackpacksWithHalfWeight(weight):
    data = open("U1.txt", "r")

    weight *= 0.5

    backpackCount = 0

    for i in range(int(next(data))):
        if(weight >= int(next(data))):
            backpackCount += 1

    data.close()

    return backpackCount


def main():
    output = open("U1rez.txt", "w")

    maxWeight = getMaxBackpackWeight()
    backpackCount = countBackpacksWithHalfWeight(maxWeight)

    output.write(f"{maxWeight} {backpackCount}")

    output.close()


if __name__ == "__main__":
    main()

File: 2016m./uzd1./sprendimai/uzd1.cpp
#include <iostream>
#include <fstream>
using namespace std;

int getMaxBackpackWeight() {
    ifstream data("U1.txt");
    int maxWeight = 0;

    int count;
    data >> count;

    for (int i = 0; i < count; ++i) {
        int weight;
        data >> weight;
        if (maxWeight < weight) {
            maxWeight = weight;
        }
    }

    data.close();
    return maxWeight;
}

int countBackpacksWithHalfWeight(int weight) {
    ifstream data("U1.txt");
    weight *= 0.5;
    int backpackCount = 0;
    int count;
    data >> count;

    for (int i = 0; i < count; ++i) {
        int currentWeight;
        data >> currentWeight;
        if (weight >= currentWeight) {
            backpackCount++;
        }
    }

    data.close();
    return backpackCount;
}

int main() {
    ofstream output("U1rez.txt");
    int maxWeight = getMaxBackpackWeight();
    int backpackCount = countBackpacksWithHalfWeight(maxWeight);

    output << maxWeight << " " << backpackCount;
    output.close();
    return 0;
}
